Assets/Scripts/Interactable.cs

```diff
@@ -18,18 +23,66 @@ void Start()
     }
 
     public void Init(){
-        this.myRenderer = GetComponent<SpriteRenderer>();
+        this.myRenderers = new List<SpriteRenderer>();
+        if (GetComponent<SpriteRenderer>() != null){
+            this.myRenderers.Add(GetComponent<SpriteRenderer>());
+        }
         this.selected = false;
+        this.cost = cost;
+        this.player = GameObject.FindGameObjectWithTag("Player").GetComponent<Player>();
+
+        this.failed = false;
+        this.waitTime = 0f;
+    }
+
+    private void LateUpdate(){
+        this.FailedPurchase();
     }
 
     public void Select(){
         this.selected = true;
-        this.myRenderer.material = selectedMaterial;
+        foreach (SpriteRenderer renderer in this.myRenderers){
+            renderer.material = selectedMaterial;
+        }
     }
 
     public void Deselect(){
         this.selected = false;
-        this.myRenderer.material = defaultMaterial;
+        foreach (SpriteRenderer renderer in this.myRenderers){
+            renderer.material = defaultMaterial;
+        }
+    }
+
+    private void FailedPurchase(){
+        if (this.failed == true){
```

Comment:
You don't need `==true` here, `this.failed` is already a boolean value
```suggestion
        if (this.failed){
```